### DISCUSSION TITLE
Anyone manage to build successfully version 19.1 with rp2 port and using WIZNET w5500 network IC? I am having undefined reference to `mod_network_nic_type_wiznet5k' error.

### DISCUSSION THREAD

**Discussion by cgtan2020** [micropython/micropython#9958] [Core Development] [answered]
[ 98%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/usbtmc/usbtmc_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/vendor/vendor_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/video/video_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/pico-sdk/src/rp2_common/pico_fix/rp2040_usb_device_enumeration/rp2040_usb_device_enumeration.c.obj
[100%] Linking CXX executable firmware.elf
CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/extmod/modnetwork.c.obj:(.rodata.mp_module_network_globals_table+0x14): undefined reference to `mod_network_nic_type_wiznet5k'
collect2: error: ld returned 1 exit status
CMakeFiles/firmware.dir/build.make:5492: recipe for target 'firmware.elf' failed
make[3]: *** [firmware.elf] Error 1
CMakeFiles/Makefile2:1577: recipe for target 'CMakeFiles/firmware.dir/all' failed
make[2]: *** [CMakeFiles/firmware.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make[1]: *** [all] Error 2
Makefile:26: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by andrewleech** [comment, marked as answer]
What was the build command / settings used? The `W5100S_EVB_PICO` is built as part of the Github CI here so must still work. There's also a W5500 board definition, did you use this? https://github.com/micropython/micropython/tree/master/ports/rp2/boards/W5500_EVB_PICO

**Reply by cgtan2020** [reply]
Thank you so much, manage to link with the correct configuration set as per what you have pointed out!!!

[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/usbtmc/usbtmc_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/vendor/vendor_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/tinyusb/src/class/video/video_device.c.obj
[ 99%] Building C object CMakeFiles/firmware.dir/home/cgtan/ver191/micropython-1.19.1/lib/pico-sdk/src/rp2_common/pico_fix/rp2040_usb_device_enumeration/rp2040_usb_device_enumeration.c.obj
[100%] Linking CXX executable firmware.elf
   text	   data	    bss	    dec	    hex	filename
 370720	    100	 204116	 574936	  8c5d8	/home/cgtan/ver191/micropython-1.19.1/ports/rp2/build-PICO/firmware.elf
[100%] Built target firmware
